product_brand/product_brand.py

```diff
@@ -33,6 +33,16 @@
 
 class product_brand(orm.Model):
     _name = 'product.brand'
+
+    def _get_products_count(self, cr, uid, ids, fields, arg, context=None):
```

Comment:
Use new API, that will avoid a lot of code. You will need to switch full class, but it deserves the effort.